### ISSUE TITLE
load testing websockets makes traefik unresponsive

### ISSUE THREAD

**Issue by tzapu** [containous/traefik#1206]
### What version of Traefik are you using (`traefik version`)?
v1.2.0-rc1

### What is your environment & configuration (arguments, toml...)?
docker service in swarm mode 
```shell
docker service create \
    --name traefik \
    --constraint 'node.role == manager' \
    --publish 80:80 --publish 8080:8080 \
    --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
    --network extracog-net \
    traefik:morbier \
    --docker \
    --docker.swarmmode \
    --docker.domain=traefik \
    --docker.watch \
    --web \
    -l DEBUG \
    --logLevel=DEBUG
```
```shell
docker service create --replicas 1 --network extracog-net --name extracog \
--constraint 'node.labels.extracog.app == true' \
--label traefik.backend.loadbalancer.sticky=true \
--label traefik.frontend.rule=Host:extracog.local \
--label traefik.frontend.entryPoints=http,https,ws,wss \
--label traefik.port=80 \
--env MONGO_OPLOG_URL=mongodb://mongo:27017/local \
--env MONGO_URL=mongodb://mongo:27017/meteor \
--env SERVER_IDENTIFIER=local \
--env PORT=80 \
localhost:5000/extracog:0.0.1 node main.js
```
### What did you do?
The service is a Meteor app that communicates through websockets/ddp.
I have load tested it with meteor-down and after a few hundred connections traefik on websockets goes totally unresponsive. Can not load anything else until i scale services (both) to 0 and back up.

Load testing on pure http endpoints using apache benchmark worked without hickups.

Removed traefik and served the service directly on port 80, load tested with meteor down again and it worked fine.

Tried with `traefik.backend.loadbalancer.sticky=false` and `traefik.backend.loadbalancer.swarm=true`, same thing.

Not sure what else to try, thanks for any insight.

**Comment by valentinvieriu** [comment]
I'm facing the same issue. Websocket traffic does not pass to my server.
Do we need a line in the toml file for websocket?
```
defaultEntryPoints = ["http","ws"]
InsecureSkipVerify = true
[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.ws]
  address = ":80"
```

( I"ve tried this too and still websockets gives me 500)

**Comment by tzapu** [comment]
just to add some more details, i reworked the config, tooke traefik out of the overlay network and used a simple backend like 
```
# traefik.toml
logLevel = "DEBUG"
defaultEntryPoints = ["http", "https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"

[file]

# rules
[backends]
  [backends.backend1]
    [backends.backend1.servers.server1]
    url = "http://172.0.0.1:88"
    weight = 10
  
[frontends]
  [frontends.frontend1]
  backend = "backend1"
  passHostHeader = true
  priority = 10
```
pointing to the docker swarm.

it works a lot better. websockets still error at some point during the benchmark but it takes a lot more effort.

the other and a lot better/more obvious thing is that traefik in this configuration does not completely die on me. i can restart benchmarks/requests through it immediately after it errors.

i have to reiterate, normal http traffic is rock solid and works without issues, this seems more websockets related. also, running the same benchmark directly on port 88, bypassig traefik yields no errors whatsoever, it just works and does not fail.

thank you for this wonderful and modern proxy, it s really cool and i hope to be able to really use it

**Comment by Vincoll** [comment]
### What version of Traefik are you using (traefik version)?
v1.2.0-rc**2**
### What is your environment & configuration (arguments, toml...)?
Rancher
Rancher Label: traefik.frontend.passHostHeader=true  / traefik.frontend.rule=Host:fqdn

```
defaultEntryPoints = ["http", "https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"


    [entryPoints.https.tls]
     [[entryPoints.https.tls.certificates]]
[...]
[rancher]
[...]
```

### What did you expect to see?
A working websocket connection. 101

### What did you see instead?
500 — Internal Server Error ( Traefik log and brower )

Content-Length: 21
Content-Type: text/plain; charset=utf-8
Date: Fri, 03 Mar 2017 20:29:00 GMT

**Comment by perlambaek** [comment]
I am seeing kinda the same thing, a lot of intermittent http 500 when proxying websockets.

**Comment by ssboisen** [comment]
Any update on this? I'm investigating Træfik as an alternative to nginx and need rock solid support for websockets

**Comment by ldez** [comment]
Do you still have the same problem with the latest version? [releases](https://github.com/containous/traefik/releases)

**Comment by Vincoll** [comment]
Problem have been fixed since 1.2.X.

**Comment by timoreimann** [comment]
thanks for the feedback, @Vincoll. 👏
